cmd/kube-proxy/app/server_test.go

```diff
@@ -403,8 +407,9 @@ func TestProcessHostnameOverrideFlag(t *testing.T) {
 			options.hostnameOverride = tc.hostnameOverrideFlag
 
 			err := options.processHostnameOverrideFlag()
-			assert.NoError(t, err, "unexpected error %v", err)
-			if tc.expectedHostname != options.config.HostnameOverride {
+			if err != nil{
+				t.Logf("space hostname is invalid")
```

Comment:
This will make it accept *any* error from `processHostnameOverrideFlag`, for any of the three test cases. You should add an `expectError` field or something, to the `testCases` struct.

Reply:
right, thanks for your suggestion, i'll do that as you say later.